empress/topoSort/test_ReconBuilder.py

```diff
@@ -0,0 +1,150 @@
+        ('m1', 'm2'): ('m1', 'm2', None, None),
+        ('m1', 'm3'): ('m1', 'm3', None, None),
+        ('m4', 'm5'): ('m4', 'm5', None, None),
+        ('m4', 'm6'): ('m4', 'm6', None, None)}
+
+        parasite_tree = { 'pTop': ('Top', 'n0', ('n0', 'n1'), ('n0', 'n5')),
+        ('n0', 'n1'): ('n0', 'n1', ('n1', 'n2'), ('n1', 'n3')),
+        ('n1', 'n3'): ('n1', 'n3', ('n3', 'n4'), ('n3', 'n6')),
+        ('n1', 'n2'): ('n1', 'n2', None, None),
+        ('n3', 'n4'): ('n3', 'n4', None, None),
+        ('n0', 'n5'): ('n0', 'n5', None, None),
+        ('n3', 'n6'): ('n3', 'n6', None, None),}
+
+        reconciliation = { ('n0', 'm1'): [('T', ('n1', 'm1'), ('n5', 'm5'))],
+        ('n1', 'm1'): [('L', ('n1', 'm2'), (None, None))],
+        ('n1', 'm2'): [('T', ('n2', 'm2'), ('n3', 'm4'))],
+        ('n2', 'm2'): [('C', (None, None), (None, None))],
+        ('n4', 'm5'): [('C', (None, None), (None, None))],
+        ('n5', 'm5'): [('C', (None, None), (None, None))],
+        ('n6', 'm6'): [('C', (None, None), (None, None))],
+        ('n3', 'm4'): [('S', ('n4', 'm5'), ('n6', 'm6'))]}
+
+        ordering_dict = TestReconBuilder.build_ordering_dict(host_tree, parasite_tree, reconciliation)
+        assert(ordering_dict == None)
+
+    @classmethod
+    def test_detect_inconsistency_2(self):
+        """
+        test topological_order recognizes another type of inconsistent reconciliation
+        """
+        # a reconciliation with another type of temporal inconsistency
+        host_tree = { 'hTop': ('Top', 'm0', ('m0', 'm1'), ('m0', 'm2')),
+        ('m0', 'm1'): ('m0', 'm1', ('m1', 'm3'), ('m1', 'm4')),
+        ('m1', 'm3'): ('m1', 'm3', None, None),
+        ('m1', 'm4'): ('m1', 'm4', None, None),
+        ('m0', 'm2'): ('m0', 'm2', None, None)}
+
+        parasite_tree = { 'pTop': ('Top', 'n0', ('n0', 'n4'), ('n0', 'n2')),
+        ('n0', 'n4'): ('n0', 'n4', None, None),
+        ('n0', 'n2'): ('n0', 'n2', ('n2', 'n5'), ('n2', 'n3')),
+        ('n2', 'n5'): ('n2', 'n5', None, None),
+        ('n2', 'n3'): ('n2', 'n3', ('n3', 'n1'), ('n3', 'n6')),
+        ('n3', 'n1'): ('n3', 'n1', None, None),
+        ('n3', 'n6'): ('n3', 'n6', None, None)}
+
+        reconciliation = { ('n0', 'm4'): [('T', ('n4', 'm4'), ('n2', 'm2'))],
+        ('n2', 'm2'): [('T', ('n5', 'm2'), ('n3', 'm1'))],
+        ('n3', 'm1'): [('S', ('n1', 'm3'), ('n6', 'm4'))],
+        ('n1', 'm3'): [('C', (None, None), (None, None))],
+        ('n4', 'm4'): [('C', (None, None), (None, None))],
+        ('n5', 'm2'): [('C', (None, None), (None, None))],
+        ('n6', 'm4'): [('C', (None, None), (None, None))]}
+
+        ordering_dict = TestReconBuilder.build_ordering_dict(host_tree, parasite_tree, reconciliation)
+        assert(ordering_dict == None)
+
+    @classmethod
+    def test_topological_order(self):
+        """
+        test topological_order by generating host and parasite trees of different sizes and going through
```

Comment:
Write a comment that says that most trees in this test is temporally consistent.

Reply:
Start a block comment with capital letter. "Test topological_order ..."